server/sonar-server/src/main/java/org/sonar/server/permission/PermissionTemplateService.java

```diff
@@ -70,9 +70,9 @@ public PermissionTemplateService(DbClient dbClient, PermissionIndexer permission
   }
   
   public boolean wouldUserHavePermissionWithDefaultTemplate(DbSession dbSession,
-    String organizationUuid, @Nullable Long userId, String permission, @Nullable String branch, String projectKey,
+    String organizationUuid, @Nullable Long userId, String globalPermission, @Nullable String branch, String projectKey,
     String qualifier) {
-    if (userSession.hasPermission(permission)) {
+    if (userSession.hasOrganizationPermission(organizationUuid, globalPermission)) {
```

Comment:
you're right. And indeed I have the feeling that this condition works only because it assumes that the param `userId` is the logged-in user (equals to userSession.getLogin()) and that `globalPermission` is `scan` (because it exists on both project and organization scopes).
@teryk @julienlancelot can we discuss about that?